cmd/algorand-indexer/daemon.go

```diff
@@ -65,276 +67,338 @@ var (
+	var err error
+	if cfg.indexerDataDir == "" {
+		err = fmt.Errorf("indexer data directory was not provided")
+		logger.WithError(err).Errorf("indexer data directory error, %v", err)
+		return err
+	}
+	if _, err = os.Stat(cfg.indexerDataDir); os.IsNotExist(err) {
+		err = os.Mkdir(cfg.indexerDataDir, 0755)
+		if err != nil {
+			logger.WithError(err).Errorf("indexer data directory error, %v", err)
+			return err
 		}
-		// Create the directory if it doesn't exist
-		if _, err := os.Stat(indexerDataDir); os.IsNotExist(err) {
-			err := os.Mkdir(indexerDataDir, 0755)
-			maybeFail(err, "failure while creating data directory: %v", err)
+	}
+	return err
+}
+
+func loadIndexerConfig(cfg *daemonConfig) error {
+	var err error
+	indexerConfigFound := util.FileExists(filepath.Join(cfg.indexerDataDir, autoLoadIndexerConfigName))
+	if indexerConfigFound {
+		if cfg.configFile != "" {
+			err = fmt.Errorf("indexer configuration was found in data directory (%s) as well as supplied via command line.  Only provide one",
+				filepath.Join(cfg.indexerDataDir, autoLoadIndexerConfigName))
+			logger.WithError(err)
+			return err
+		}
+		// No config file supplied via command line, auto-load it
+		configs, err := os.Open(cfg.configFile)
+		if err != nil {
+			logger.WithError(err).Errorf("%v", err)
+			return err
+		}
+		defer configs.Close()
+		err = viper.ReadConfig(configs)
+		if err != nil {
+			logger.WithError(err).Errorf("invalid config file (%s): %v", viper.ConfigFileUsed(), err)
+			return err
+		}
+	}
+	if cfg.configFile != "" {
+		configs, err := os.Open(cfg.configFile)
+		if err != nil {
+			logger.WithError(err).Errorf("File Does Not Exist Error: %v", err)
+			return err
+		}
+		defer configs.Close()
+		err = viper.ReadConfig(configs)
+		if err != nil {
+			logger.WithError(err).Errorf("invalid config file (%s): %v", viper.ConfigFileUsed(), err)
+			return err
 		}
+		logger.Infof("Using configuration file: %s\n", cfg.configFile)
+	}
+	return err
+}
 
-		// Detect the various auto-loading configs from data directory
-		indexerConfigFound := util.FileExists(filepath.Join(indexerDataDir, autoLoadIndexerConfigName))
-		paramConfigFound := util.FileExists(filepath.Join(indexerDataDir, autoLoadParameterConfigName))
-
-		// If we auto-loaded configs but a user supplied them as well, we have an error
-		if indexerConfigFound {
-			if configFile != "" {
-				logger.Errorf(
-					"indexer configuration was found in data directory (%s) as well as supplied via command line.  Only provide one.",
-					filepath.Join(indexerDataDir, autoLoadIndexerConfigName))
-				panic(exit{1})
-			}
-			// No config file supplied via command line, auto-load it
-			configs, err := os.Open(configFile)
-			if err != nil {
-				maybeFail(err, "%v", err)
-			}
-			defer configs.Close()
-			err = viper.ReadConfig(configs)
-			if err != nil {
-				maybeFail(err, "invalid config file (%s): %v", viper.ConfigFileUsed(), err)
-			}
+func loadIndexerParamConfig(cfg *daemonConfig) error {
```

Comment:
These helpers don't need the `daemonConfig`, you can pass in the specific value they need.

Reply:
Fixed--this function in particular stores the resolved variable in the config struct so I've left that one as is.